hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -104,26 +117,11 @@ void refreshPartitions() {
         }
     }
 
-    @Override
-    public void handlePartitionsEventV15(Collection<Map.Entry<Address, List<Integer>>> collection, int partitionStateVersion) {
-        processPartitionResponse(collection, partitionStateVersion, true);
-    }
-
-    @Override
-    public void beforeListenerRegister() {
-
-    }
-
-    @Override
-    public void onListenerRegister() {
-
-    }
-
     private void waitForPartitionsFetchedOnce() {
-        while (partitions.isEmpty() && client.getConnectionManager().isAlive()) {
+        while (getPartitions().isEmpty() && client.getConnectionManager().isAlive()) {
```

Comment:
`getPartitions().isEmpty()` or `partitionSateVersion == -1`?

Reply:
It feels like we broke the guarantee here. And now both seem wrong to me.